src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -504,6 +506,13 @@ export class ConnectionWidget extends lifecycle.Disposable {
 			this._tableContainer.classList.remove('hide-username');
 			this._tableContainer.classList.add('hide-password');
 			this._tableContainer.classList.remove('hide-azure-accounts');
+		} else if (currentAuthType === AuthenticationType.dSTSAuth) {
+			this._accountManagementService.getDstsToken(this.serverName, this.databaseName).then(token => {
+				this.dstsToken = token;
+			});
+			this._tableContainer.classList.add('hide-username');
+			this._tableContainer.classList.add('hide-password');
+			this._tableContainer.classList.add('hide-azure-accounts');
```

Comment:
Unfortunately the logic to control what and how UI elements are shown in the connection widget are hard coded here.

Reply:
It would really be nice to have this stuff do something like we did for resource deployment and have the JSON contribution define what fields are shown so we don't have to keep doing this. Talk to @kburtram about this kind of work.